client/landing/stepper/declarative-flow/site-migration-flow.ts

```diff
@@ -79,24 +79,8 @@ const siteMigration: Flow = {
 			}
 		}, [ isAdmin ] );
 
-		useEffect( () => {
-			// We don't need to do anything if the user isn't logged in.
-			if ( ! userIsLoggedIn ) {
-				return;
-			}
-
-			if ( siteSlug || siteId ) {
-				return;
-			}
-
-			if ( isHostedSiteMigrationFlow( flowPath ) ) {
-				return;
-			}
-
-			window.location.assign( '/start' );
-		}, [ flowPath, siteId, siteSlug, userIsLoggedIn, isAdmin ] );
-
-		if ( ! siteSlug && ! siteId && ! isHostedSiteMigrationFlow( flowPath ) ) {
+		if ( userIsLoggedIn && ! siteSlug && ! siteId && ! isHostedSiteMigrationFlow( flowPath ) ) {
+			window.location.assign( '/' );
```

Comment:
@escapemanuele I reverted to use window.location.assign instead of redirect(), since the latter uses window.location.href instead of window.location.assign(). This unnecessarily complicates the unit test setup so I thought this would be simpler.

Reply:
Thanks @taipeicoder, that makes sense 👍